xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/src/test/it/org/xwiki/flamingo/test/docker/DeletePageRecyclebinIT.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.flamingo.test.docker;
+
+import java.util.HashMap;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.xwiki.flamingo.skin.test.po.DeleteConfirmationPage;
+import org.xwiki.test.docker.junit5.UITest;
+import org.xwiki.test.ui.TestUtils;
+import org.xwiki.test.ui.po.ConfirmationPage;
+import org.xwiki.test.ui.po.DeletePageOutcomePage;
+import org.xwiki.test.ui.po.DeletingPage;
+import org.xwiki.test.ui.po.ViewPage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests the Delete Page feature.
+ *
+ * @version $Id$
+ * @since 12.8RC1
+ */
+@UITest(properties = { "xwikiPropertiesAdditionalProperties=refactoring.recyclebin.skip=true" })
+public class DeletePageRecyclebinIT
```

Comment:
Why is this a complete new test? This looks expensive. Don't we have a functional test already that verifies page delete?

Reply:
We do, but not with `refactoring.recyclebin.skip` set to `true`.

Reply:
I can possibly replace the `refactoring.recyclebin.skip` property with a configuration XClass and an entry in the administration. That way this can be changed without having to restart the server, and those tests can be merged into `DeletePageIT`. WDYT?

Reply:
> We do, but not with refactoring.recyclebin.skip set to true.

of course, but why not add a test method in the existing scenario?

Reply:
> I can possibly replace the refactoring.recyclebin.skip property with a configuration XClass and an entry in the administration. That way this can be changed without having to restart the server, and those tests can be merged into DeletePageIT. WDYT?

Indeed would be good to have it in the UI I think. Interesting how if you listent to a test you'll change your design :)

Reply:
New configuration class introduced https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-a7454e8850bc014a1b35665113c86f0dR23
But the definitive place of the configuration is still discussed, see https://forum.xwiki.org/t/xwiki-17401-option-for-deleting-documents-and-or-attachments-without-sending-to-recycle-bin/7340/22